ArduSub/Sub.cpp

```diff
@@ -13,14 +13,28 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 #include "Sub.h"
+
+#define FORCE_VERSION_H_INCLUDE
 #include "version.h"
+#undef FORCE_VERSION_H_INCLUDE
 
 const AP_HAL::HAL& hal = AP_HAL::get_HAL();
 
 /*
   constructor for main Sub class
  */
-Sub::Sub(void) :
+Sub::Sub(void)
+    : fwver{
```

Comment:
Why are we constructing this here now?

Not that this should be `static constexpr`, not just `const` (my bad)

Reply:
It's ok to put on header when you don't need to bring as dependency the entire project as a result.

Initialization lists have their place and should not be banned because of style. Yes, it's preferable to use c++11 initialization, but don't do this a rule with no exceptions. Exception here is very much justified.